.github/workflows/deploy_release_to_marketplace.yml

```diff
@@ -0,0 +1,48 @@
+name: autokittehVSCode CI - Publish to VSCode Marketplace
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        type: choice
+        description: "Environment to deploy to"
+        options:
+          - "pre-release"
+          - "release"
+        required: true
+      tag:
+        type: string
+        description: "Git tag to deploy"
+        required: true
+
+permissions:
+  contents: write
+  repository-projects: write
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.tag }}
+          token: ${{ secrets.PAT_TOKEN }}
+          submodules: true
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "21.2.0"
+          cache: "npm"
+      - name: Pull submodule from remote
+        run: git submodule update --remote
+      - name: Install dependencies for both vscode and react app
+        run: npm ci
+      - name: Compile vscode and react app
+        run: npm run compile:all
+      - name: Build vscode and react app
```

Comment:
Any chance that you take an artifact you already checked before and not rebuilding from scratch when you create a release? 

For pre release you should build from scratch